File: DSA/Ayush_Mishra_590014844/Day_16_Question_1.c
#include <stdio.h>

int linear_search(int arr[], int size, int target) {
    for (int i = 0; i < size; i++) {
        if (arr[i] == target)
            return i;
    }
    return -1;
}

int main() {
    int n, target;
    printf("Enter number of elements: ");
    scanf("%d", &n);
    int arr[n];

    printf("Enter %d elements: ", n);
    for (int i = 0; i < n; i++)
        scanf("%d", &arr[i]);

    printf("Enter target element: ");
    scanf("%d", &target);

    int result = linear_search(arr, n, target);
    if (result != -1)
        printf("Element found at index: %d\n", result);
    else
        printf("Element not found.\n");

    return 0;
}

File: DSA/Ayush_Mishra_590014844/Day_16_Question_1.py
def linear_search(arr, target):
    for i in range(len(arr)):
        if arr[i] == target:
            return i
    return -1

# Input from user
arr = list(map(int, input("Enter array elements separated by space: ").split()))
target = int(input("Enter the target element: "))
result = linear_search(arr, target)
print(f"Element found at index: {result}" if result != -1 else "Element not found.")
